safekeeper/src/pull_timeline.rs

```diff
@@ -465,14 +465,27 @@ pub async fn handle_request(
     assert!(status.tenant_id == request.tenant_id);
     assert!(status.timeline_id == request.timeline_id);
 
-    pull_timeline(
+    match pull_timeline(
         status,
         safekeeper_host,
         sk_auth_token,
         http_client,
         global_timelines,
     )
     .await
+    {
+        Ok(resp) => Ok(resp),
+        Err(e) => {
+            if let Some(TimelineError::AlreadyExists(_) | TimelineError::CreationInProgress(_)) =
```

Comment:
`CreationInProgress` does not guarantee that the parallel creation will succeed, so returning `200 OK` is not ok for it

Reply:
Fair point. We could return `202 Accepted`, or `409 Conflict`. What do you think? I'm leaning towards latter.

Reply:
Yeah, 409 looks better to me too

Reply:
Done